### PR TITLE
GPL refactor

### PR DESCRIPTION
**Pull request by kotfic**
refactor ansible running portion of the code to be compliant with GPL

### CODE REVIEW

**Comment by kotfic** [comment]
@cjh1  PTAL

This refactors run_playbook to work with Popen rather than calling ansible directly through the pythonic API. Something to consider is if we want to use a library like [pexpect](https://pexpect.readthedocs.org/en/stable/) instead to simulate a pty and scrape output for errors. 

This also implements a fair amount of novel code around modeling an ansible inventory.  Currently we're just using it to generate a simple inventory for localhost,  but provisioning may need a more complex inventory so i put together code/tests for a more feature complete inventory.

File: cumulus/ansible/tasks/cluster.py [hunk 1]
```diff
@@ -12,24 +16,25 @@ def get_playbook_path(name):
...
-        extra_vars=extra_vars
-    )
-    # Note:  can refer to callback.playbook.extra_vars  to get access
-    # to girder_token after this point
-    results = pb.run()
+    cmd.append(playbook)
 
-    return results
+    p = subprocess.Popen(cmd, env=env, stdout=subprocess.PIPE)
+
```
Comment (cjh1) [inline cumulus/ansible/tasks/cluster.py]: what about stderr?

File: cumulus/ansible/tasks/cluster.py [hunk 2]
```diff
@@ -12,24 +16,25 @@ def get_playbook_path(name):
...
+    for line in iter(p.stdout.readline, b''):
+        logger.info(line)
```
Comment (cjh1) [inline cumulus/ansible/tasks/cluster.py]: As discussed, we should also check the exit code here.

File: cumulus/ansible/tasks/cluster.py (hunk 1 above)
Comment (kotfic) [inline cumulus/ansible/tasks/cluster.py]: @cjh1 we can grab stderr,  but this is just for output to the celery log. First,  is it nessisary? If so - If we pull it,  should we log it before or after stdout?

File: cumulus/ansible/tasks/cluster.py (hunk 2 above)
Comment (kotfic) [inline cumulus/ansible/tasks/cluster.py]: This is implemented in https://github.com/Kitware/cumulus/pull/176

File: cumulus/ansible/tasks/cluster.py (hunk 1 above)
Comment (cjh1) [inline cumulus/ansible/tasks/cluster.py]: Its necessary if we think ansible has failure modes that are only observable through stderr or important diagnostic information is written there, my feeling it wouldn't hurt to add it. Order not important to me, I would write the stderr entry at the ERROR level.

File: cumulus/ansible/tasks/cluster.py (hunk 2 above)
Comment (cjh1) [inline cumulus/ansible/tasks/cluster.py]: Excellent

**Comment by kotfic** [comment]
@cjh1 this is ready for review.  thanks!

**Comment by cjh1** [comment]
One minor nit, can we use single quotes for strings? Other than that LGTM

**Comment by cjh1** [comment]
LGTM

**Pull request merged**